CorporateHotelBooking/CorporateHotelBooking.Test/E2E/CompanyServiceAcceptanceTest.cs

```diff
@@ -31,14 +31,15 @@ public async void should_be_able_to_add_employee()
 
             // Assert
             Assert.Equal(HttpStatusCode.Created, addEmployeeResponse.StatusCode);
+            
+            // Just and example that could help preventing the access to the repo from
+            // the acceptance test.
+            // The treadoff here is that I am assuming that we will at some point in time have a get endpoint anyways.
+            // So given that this is true, this acceptance test would then test even more the outside-part, then
+            // in the inside you can actually assert the repo.
+            var listEmployeeResponse = await _client.GetAsJsonAsync("employees", new { CompanyId = companyId, EmployeeId = employeeId });
 
-            ThenEmployeeShouldExistInRepository();
-
-            void ThenEmployeeShouldExistInRepository()
-            {
-                var employeeRepository = _apiFactory.Services.GetService<IEmployeeRepository>();
-                Assert.NotNull(employeeRepository.Get(EmployeeId.From(employeeId)));
-            }
+            Assert.NotNull(listEmployeeResponse.Get(EmployeeId.From(employeeId)));
```

Comment:
to add more info on this end:

The assumption is that the feature will require the get anyways, so you don't have to put the endpoint just for testing.

Usually in the acceptance test different workflows and use cases are triggered, for example:

Create a hotel first
book a room
Make a reservation for a given room
checkout
This is like a flow that makes sense from the business point of view.

In an acceptance test, if you want to create this flow probably you will have different endpoints to do so.

If that is true, the acceptance test will become the happy flow assertion the checkpoints for each step. So for example: to book a room you first need a hotel, so when the time comes to create a room and the hotel si not there the test will most likely to fail saying "hey I cannot find the hotel".

In that sense, you will will not need to use the repo to make sure the info is In the database (be it in memory or SQL or whatever).

The e2e test will give the feedback that it cannot find the hotel and now we need to figure out in which part of the app this is happening.